python/fledge/plugins/filter/rename/rename.py

```diff
@@ -166,14 +166,31 @@ def find_and_replace(operation, find, replace_with, ignore_case, reading):
         dict:          A processed dictionary
     """
     revised_reading_dict = {}
-    _flag = re.IGNORECASE if ignore_case == 'false' else False
-
-    def get_all_values(nested_dictionary):
-        for key, value in nested_dictionary.items():
-            if type(value) is dict:
-                get_all_values(value)
-            _datapoint = re.sub(search_pattern, replace_with, str(key), flags=_flag)
-            revised_reading_dict[_datapoint] = value
```

Comment:
I believe here simple ELSE will work - means only update dictionary of value is non-dict type. As recursive function caller is there for dict type

Reply:
As discussed, only adding else doesn't work.